registry/remote/repository.go

```diff
@@ -944,46 +945,99 @@ func (s *manifestStore) FetchReference(ctx context.Context, reference string) (d
 }
 
 // generateDescriptor returns a descriptor generated from the response.
-func (s *manifestStore) generateDescriptor(resp *http.Response, ref registry.Reference) (ocispec.Descriptor, error) {
+func (s *manifestStore) generateDescriptor(resp *http.Response, ref registry.Reference, calculateDigest bool) (ocispec.Descriptor, error) {
```

Comment:
When do we need to calculate the digest, and when do we not? Is it based on the request method?

Reply:
Yes, based on the request method.  In the new revision (not yet published), I've updated `generateDescriptor` to take not the `bool calculateDigest`, but instead `string reqMethod`; then we do the "calculation" inside `generateDescriptor` and not have the caller make any decisions.